linux_pvstate/driver_deps.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * driver_deps.c    Propagate MSTI port states to Linux Per VLAN STP states
+ *
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; either version
+ *  2 of the License, or (at your option) any later version.
+ *
+ * Authors: Vitalii Demianets <dvitasgs@gmail.com>
+ * Authors: Jonas Gorski <jonas.gorski@bisdn.de> -- Set Linux Per VLAN state
+ */
+
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <string.h>
+#include <asm/byteorder.h>
+
+#include <linux/if_bridge.h>
+
+#include "bridge_ctl.h"
+#include "libnetlink.h"
+#include "log.h"
+#include "mstp.h"
+
+static int br_set_vlan_state(struct rtnl_handle *rth, unsigned ifindex, __u16 vid, __u8 state)
```

Comment:
i'd merge this per-VLAN state code into the main driver_deps code;
maybe guarded via a global flag here;

`driver_set_per_vlan_state(bool enabled) `   or something like that

otherwise looks good;

Reply:
Actually I completely merged this into the main mstpd (well bridge*) code, since just setting the per-vlan state isn't enough.